### ISSUE TITLE
[BUG] Get-PnPFile fails to download multiple small files in a row (SharePoint Online)

### ISSUE THREAD

**Issue by SMatPQ** [pnp/powershell#1618]
### Reporting an Issue or Missing Feature
Trouble downloading multiple small files in a row with Get-PnPFile.
Wether it's the same file or different ones.

Seems to affect the current types of file : 
- xml
- txt

Might be related to the following commit since **it's working just fine in version 1.8.0**
fe7c1615fb394ca052ad30c5a4ac48d402762344

### Expected behavior 
To download all the files.

### Actual behavior 
The first two files get downloaded without any problem.
For the third file, the command hangs for a minute and the file doesn't get downloaded as if the it doesn't exist on the SharePoint library.
**Throws this error:**

Get-PnPFile : One or more errors occured.
At line:1 char:1
 Get-PnPFile -Url "/sites/Laboratoire/Documents partages/WW-1498680-A- ...
     CategoryInfo          : WriteError: (:) [Get-PnPFile], AggregateException
     FullyQualifiedErrorId : EXCEPTION,PnP.PowerShell.Commands.Files.GetFile

![image](https://user-images.githubusercontent.com/99362659/153290143-8e33a0a5-4536-4ef3-a859-00e884fa9bf2.png)

### Steps to reproduce behavior
`Get-PnPFile -Url "/sites/Laboratoire/Documents partages/WW-1498680-A-270634-2204994-0-AdvStd-33-P-20-0.xml" -Path 'C:\Users\username\Desktop' -FileName 'testpnp.xml' -AsFile -Force`

!**Do the above command three times in a row**!

### What is the version of the Cmdlet module you are running?
Version 1.9.0

### Which operating system/environment are you running PnP PowerShell on?
- [ X] Windows 10, Powershell 5.1 with SharePoint Online

**Comment by gautamdsheth** [comment]
Can you try with the latest nightly build ? 
Was not able to reproduce with the steps that you mentioned.

**Comment by SMatPQ** [comment]
After some testing :

- **Same issue with 1.9.41-nightly build, with PowerShell 5.1**
- Works great with 1.9.41-nightly w/ PowerShell 7.2.1
- Works great with 1.9.0 stable w/ PowerShell 7.2.1

It's worth noting that my main bug report was with PowerShell 5.1

I will keep testing and update here along the way.

**Comment by SamLedFR** [comment]
Hi,

Not really same issue cause I didn't used **-Force**, but faced weird error. Hope it will probably help you to understand what's going wrong underneath.

In my script I was trying to download a file locally from one tenant to upload it to another one. It was working like a charm when suddenly throw same error. Got full stack error with Get-Error (see attached) :

The WriteObject and WriteError methods cannot be called from outside the overrides of the
BeginProcessing, ProcessRecord, and EndProcessing methods, and they can only be called from within the same thread.
Validate that the cmdlet makes these calls correctly, or contact Microsoft Customer Support Services.

Downloaded sources and had a look at line 85 in GetFile.cs
```
if (!Force)
{
      `WriteWarning($"File` '{fileToSave}' exists already. use the -Force parameter to overwrite the file.");
}
```

I added -Force and it's now working as expected. Seems that WriteWarning call generates an unexpected behavior.

Versions used : PnP 1.9.0 on PS 5.1 and 7.2.1 (same behavior)

Regards,
Sam

[Get-Error.txt](https://github.com/pnp/powershell/files/8044712/Get-Error.txt)

**Comment by JazBInKC** [comment]
Having a similar (but different) issue.  Get-PnPFile writes (for example) 2 or 3 'new' files to NTFS and then gakks on the 4th with the "several errors occurred" message.   I'm using -Force and the behavior is not rectified by using the latest (1.10) release (under PS 5.1).   After the error, Get-PnPFile may write files or may continue to error.  Killing the process and re-running often allows the 'error' files to write, and then sometimes not.  Also - I'm checking for the presence of the target before writing I'm only writing new or updated files to the target.

jbw

**Comment by JazBInKC** [comment]
Interesting.  My issue WAS resolved by going to PS 7.2.2 with 1.10.0 PnP.PowerShell.  I'm copying 15 million files to NTFS - the hang time with the failing Get-PnPFile calls was killing me - and the error messages were of no help in diagnosing the actual cause of failure.

**Comment by gautamdsheth** [comment]
Managed to find a fix for the PS 5 issue, fix has been merged. 
Will be available in tomorrow's nightly.
Please let me know how it goes and feel free to reopen the issue again if the issue persists.

Do remember to clear the old installations of PnP PowerShell before you test it out to avoid any weird behaviors.
Thanks for raising this issue :)
